apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/source/RdaVersion.java

```diff
@@ -0,0 +1,181 @@
+package gov.cms.bfd.pipeline.rda.grpc.source;
+
+/** A utility class for comparing version numbers. */
+public class RdaVersion {
+
+  /** The major version (first number). */
+  private final int major;
+  /** The minor version (middle number). */
+  private final int minor;
+  /** The patch version (last number). */
+  private final int patch;
+
+  /**
+   * The required {@link CompatabilityLevel}, only used when comparing another {@link RdaVersion} to
+   * this one.
+   */
+  private final CompatabilityLevel compatability;
+
+  /**
+   * Builds an {@link RdaVersion} instance using the given version specifications.
+   *
+   * @param major The major version
+   * @param minor The minor version
+   * @param patch The patch version
+   * @param compatability The level requirement
+   */
+  private RdaVersion(int major, int minor, int patch, CompatabilityLevel compatability) {
+    this.major = major;
+    this.minor = minor;
+    this.patch = patch;
+    this.compatability = compatability;
+  }
+
+  /**
+   * Creates a {@link RdaVersionBuilder} to be used to build an {@link RdaVersion} object.
+   *
+   * @return A {@link RdaVersionBuilder}.
+   */
+  public static RdaVersionBuilder builder() {
+    return new RdaVersionBuilder();
+  }
+
+  /**
+   * Determines if this {@link RdaVersion} specifies a version or range that would accept the given
+   * {@link RdaVersion}.
+   *
+   * @param rdaVersionString The version being checked to see if it falls within range.
+   * @return True if the given {@link RdaVersion} is within range of this instance's {@link
+   *     RdaVersion} requirement.
+   */
+  public boolean allows(String rdaVersionString) {
+    // When we connect to S3 or some other source, there is a prefix on the version to distinguish
+    // it.
+    String[] bits = rdaVersionString.split(":");
+    String versionString = bits[bits.length - 1];
+    // Parse the version
+    RdaVersion rdaVersion = builder().versionString(versionString).build();
+
+    boolean compatable;
+
+    switch (compatability) {
+      case MAJOR:
+        compatable = rdaVersion.major == major;
+        break;
+      case MINOR:
+        compatable = rdaVersion.major == major && rdaVersion.minor == minor;
+        break;
+      default:
+        compatable =
+            rdaVersion.major == major && rdaVersion.minor == minor && rdaVersion.patch == patch;
+    }
```

Comment:
Not sure this logic matches the NPM semver spec.

https://semver.npmjs.com

For example `~` says: "include everything greater than a particular version in the same minor range" whereas with this logic `~0.10.3` would accept `0.10.2`.  I think for MINOR we should add `rdaVersion.patch >= patch`.

Reply:
MAJOR should probably do the same kind of thing too I think.  Check both minor and patch to be `>=` the set value.